File: src/main/java/istu/bacs/model/LanguageConverter.java
package istu.bacs.model;

import org.springframework.stereotype.Component;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Component
@Converter(autoApply = true)
public class LanguageConverter implements AttributeConverter<Language, Integer> {
    @Override
    public Integer convertToDatabaseColumn(Language language) {
        return language.getLanguageId();
    }

    @Override
    public Language convertToEntityAttribute(Integer languageId) {
        return Language.findById(languageId);
    }
}

File: src/main/java/istu/bacs/model/LanguageConverter.kt
package istu.bacs.model

import org.springframework.stereotype.Component
import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Component
@Converter(autoApply = true)
class LanguageConverter : AttributeConverter<Language, Int> {
    override fun convertToDatabaseColumn(language: Language?) = language!!.languageId
    override fun convertToEntityAttribute(id: Int?) = Language.findById(id!!)
}